modules/navier_stokes/test/tests/finite_volume/wcns/materials/1d_test_cpT.i

```diff
@@ -0,0 +1,231 @@
+L = 30
+bulk_u = 0.01
+p_ref = 101325.0
+T_in = 860
+q_source = 50000
+
+A_cp = 976.78
+B_cp = 1.0634
+rho = 2000
+
+advected_interp_method = 'upwind'
+
+[Mesh]
+  [gmg]
+    type = GeneratedMeshGenerator
+    dim = 1
+    xmin = 0
+    xmax = ${L}
+    nx = 10
+  []
+[]
+
+[GlobalParams]
+  rhie_chow_user_object = 'rc'
+  advected_interp_method = ${advected_interp_method}
+  velocity_interp_method = 'rc'
+[]
+
+[UserObjects]
+  [rc]
+    type = INSFVRhieChowInterpolator
+    u = vel_x
+    pressure = pressure
+  []
+[]
+
+[Variables]
+  [vel_x]
+    type = INSFVVelocityVariable
+    initial_condition = ${bulk_u}
+    two_term_boundary_expansion = false
+  []
+  [pressure]
+    type = INSFVPressureVariable
+    initial_condition = ${p_ref}
+    two_term_boundary_expansion = false
+  []
+  [T]
+    type = INSFVEnergyVariable
+    two_term_boundary_expansion = false
+    initial_condition = ${T_in}
+  []
+[]
+
+[FVKernels]
+  [mass]
+    type = WCNSFVMassAdvection
+    variable = pressure
+    rho = 'rho'
+  []
+
+  [u_advection]
+    type = INSFVMomentumAdvection
+    variable = vel_x
+    rho = 'rho'
+    momentum_component = 'x'
+  []
+  [u_viscosity]
+    type = INSFVMomentumDiffusion
+    variable = vel_x
+    mu = 'mu'
+    momentum_component = 'x'
+  []
+  [u_pressure]
+    type = INSFVMomentumPressure
+    variable = vel_x
+    momentum_component = 'x'
+    pressure = pressure
+  []
+
+  [temp_conduction]
+    type = FVDiffusion
+    coeff = 'k'
+    variable = T
+  []
+  [temp_advection]
+    type = INSFVEnergyAdvection
+    variable = T
+  []
+  [source]
+    type = FVBodyForce
+    variable = T
+    function = source_func
+  []
+[]
+
+[FVBCs]
+  [inlet-u]
+    type = INSFVInletVelocityBC
+    boundary = 'left'
+    variable = vel_x
+    functor = ${bulk_u}
+  []
+  [inlet_T]
+    type = FVDirichletBC
+    variable = T
+    boundary = 'left'
+    value = ${T_in}
+  []
+
+  [outlet_p]
+    type = INSFVOutletPressureBC
+    boundary = 'right'
+    variable = pressure
+    function = ${p_ref}
+  []
+[]
+
+[Functions]
+  [source_func]
+    type = ParsedFunction
+    expression = '${q_source}'
+  []
+[]
+
+[FunctorMaterials]
+  [converter_to_regular_T]
+    type = FunctorADConverter
+    ad_props_in = 'T'
+    reg_props_out = 'T_nAD'
+  []
+  [ins_fv]
+    type = INSFVEnthalpyFunctorMaterial
+    temperature = 'T'
+    rho = 'rho'
+    cp = 'cp'
+
+    assume_constant_cp = false
+    h_in = 'h'
+    # fp = 'fp'
+    # pressure = 'pressure'
```

Comment:
Remove is unused

Reply:
it's the other way of setting this up. I ll add a comment